### ISSUE TITLE
Unable to change text alignment in text input

### ISSUE THREAD

**Issue by Jaiieq** [livewire/flux#552]
```blade
<flux:input label="Funding" type="text" value="$12,000.00" class="!text-end" />
```

The above code snippet produces this:

![Image](https://github.com/user-attachments/assets/6231ad65-a575-457f-9519-b9492e7edfad)

...when I want it to produce this:

![Image](https://github.com/user-attachments/assets/7a9952e6-dad4-434e-8f62-fe8330368e7c)

Am I missing something, is this an oversight in flux, or am I just being dumb?  (Or a combination of the three!).  It seems the text can only be left aligned.

**Comment by igmltd** [comment]
<flux:input label="Funding" type="text" value="$12,000.00" class="[&>input]:text-right" />

This seems to work

**Comment by jeffchown** [comment]
Nice @igmltd - thanks.

**Comment by Jaiieq** [comment]
That worked!  Much appreciated @igmltd!  I would have never thought to do that.

**Comment by calebporzio** [comment]
Yeah this is a tricky one because there are two elements inside the `<flux:input>` component:

```html
<div> <!-- A wrapper div -->
    <input ... > <!-- The actual input element -->
</div>
```

Currently, `class` and `style` attributes are applied to the wrapping div and everything else is applied to the input element. (Read more about that here: https://fluxui.dev/docs/patterns#split-attribute-forwarding)

The reasoning is that it's more common to customize a flux component's layout/spacing using classes than it's internal style (like in this case with text alignment) (Read more about that here: https://fluxui.dev/docs/principles#we-style,-you-space)

---

Workarounds:
* Complex tailwind utilities like `[&>input]:text-end` to target the nested element
  * Not ideal because it feels hacky and requires knowledge of component internals
* Some bonkers algorithm that splits out classes that should be applied to the wrapping div and ones that should be applied to the input
  * Never going to happen but worth mentioning for posterity

---

Potential non-workaround solutions:

A) We just tell people to publish the `<flux:input>` component and add a special prop called `text-align` or something

B) We add the `text-align` prop ourselves

C) We expose a slot as a way to apply classes to the inner element like so:

```html
<flux:input>
    <x-slot name="input" class="text-end"></x-slot>
</flux:input>
```

This is kinda clever. We could use the `{{ $input->attributes }}` inside the flux input component as a catcher of attributes to forward to the nested input.

However, it does not read well/is not obvious what it does. Also, it turns one line into three.

D) We expose a `input-class` prop or something to specify when you want classes to go on the input element itself.

Again, I don't really like this because it feels hacky and the prop name `input-class` doesn't feel great.

---

After writing this out, I think for now, selectors like `[&>input]:?` are the least regrettable. Let's see if others keep running into this, and if so, maybe we just add a bit to the documentation explaining this pattern.

Thanks for the report, I'm going to close this for now.

**Comment by Jaiieq** [comment]
Thanks @calebporzio, I appreciate the explanation.

I think it may be worth putting the tailwind method `[&>input]:?` in the documentation, at least until there is a new potential solution or better workaround, if one exists!  I'm just being devils advocate here as maybe not everyone would think to search the bug reports and just assume "This is just how the component works."  But I also understand this is likely a small use case.

**Comment by jeffchown** [comment]
> C) We expose a slot as a way to apply classes to the inner element like so:

That one is kind of cool, but it isn't as concise when not needed.

I agree that putting the `[&>input]:?` in the documentation would be a good idea for now as it is the best solution that currently exists.

**Comment by edalzell** [comment]
I just started to look at this and would like to override the default input styling. I was hoping not to publish the components cuz they're complicated.

@calebporzio why can't we have a prop to pass? like `inputClass` or something?
